plugins/destination/postgresql/client/list_tables.go

```diff
@@ -127,5 +127,7 @@ func (c *Client) listTables(ctx context.Context) (schema.Tables, error) {
 			Unique: isUnique && uniqueName == tableName+"_"+columnName+"_key",
 		})
 	}
+
+	c.pgTablesToPKConstraints = tableInfo
```

Comment:
Yeah let's add a lock here

Reply:
Added in https://github.com/cloudquery/cloudquery/pull/17337/commits/358cd4f8e7f5e74b9209b31d5b9a8be432bc6a74